jpsonic-main/src/main/java/com/tesshu/jpsonic/domain/JapaneseReadingUtils.java

```diff
@@ -382,13 +381,25 @@ private ReadingResult toReading(Token token) {
         String result = s;
         for (String article : settingsService.getIgnoredArticlesAsArray()) {
             if (StringUtils.startsWithIgnoreCase(s, article + SPACE)) {
-                // reading = lower.substring(article.length() + 1) + ", " + article;
                 result = result.substring(article.length() + 1);
             }
         }
         return result;
     }
 
+    private @Nullable String rotateArticles(@Nullable String s) {
+        if (isEmpty(s)) {
+            return null;
+        }
+        String result = s;
+        for (String article : settingsService.getIgnoredArticlesAsArray()) {
+            if (StringUtils.startsWithIgnoreCase(s, article + SPACE)) {
+                result = result.substring(article.length() + 1) + ", " + article;
+            }
+        }
+        return result;
+    }
+
```

Comment:
Consideration of rules regarding article removal. Rotation is sometimes seen as a method of processing articles that gets in the way when sorting in Latin. In that method, all entries are converted to "The Beatles -> Beatles, The" and then sorted. Personally, I don't think it's a very meaningful cost, and I think it would be better to simply delete it. (We cannot imagine any realistic case where this difference would cause a problem. Of course, if such cases are discovered in the future, they can be fixed at that time.)

Newly created sorts in Jpsonic are already unified to simple deletion. Rotation processing has been maintained to preserve Airsonic's implementation, but there is no need for rotation when generating Music-index. It looks like the string rotation process can now be removed.

Reply:
I think that the existence of this kind of logic is probably a remnant of Pgrammer's classification done at a primitive stage, asking whether the article was deleted or whether the string was originally a string without an article. There is a high possibility that this is one of the bad habits rather than clean logic. (By the way, Japanese always uses Non-article noun, so maybe I'm just insensitive to this issue and it just looks like debug code.)

Reply:
Done.